From e25442709e0aa96a137230980fe75bd003b4094b Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelman@morgridge.org>
Date: Tue, 25 Jul 2023 12:13:25 -0500
Subject: [PATCH] Fix logic error in user mapping

When setting up mapping rules, if the user feature was utilized, a
logic error would cause the rule to be ignored.

Fixes #2056
---
 src/XrdSciTokens/XrdSciTokensAccess.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XrdSciTokens/XrdSciTokensAccess.cc b/src/XrdSciTokens/XrdSciTokensAccess.cc
index 74c2797bf..25b3f9ecc 100644
--- a/src/XrdSciTokens/XrdSciTokensAccess.cc
+++ b/src/XrdSciTokens/XrdSciTokensAccess.cc
@@ -239,7 +239,7 @@ struct MapRule
     {
         if (!m_sub.empty() && sub != m_sub) {return "";}
 
-        if (!m_username.empty() && username != username) {return "";}
+        if (!m_username.empty() && username != m_username) {return "";}
 
         if (!m_path_prefix.empty() &&
             strncmp(req_path.c_str(), m_path_prefix.c_str(), m_path_prefix.size()))
-- 
2.34.1

